iceoryx_posh/test/moduletests/test_popo_server_port_roudi.cpp

```diff
@@ -0,0 +1,262 @@
+}
+
+TEST_F(ServerPort_test, ReleaseAllChunksWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f29f8890-c106-470d-820f-32eeea694f17");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    // produce chunks for the chunk receiver
+    pushRequests(sut.requestQueuePusher, QUEUE_CAPACITY);
+
+    // produce chunks for the chunk sender
+    allocateResponseWithRequestHeaderAndThen(sut, [&](const auto, auto res) { sut.portUser.sendResponse(res); });
+
+    EXPECT_THAT(getNumberOfUsedChunks(), Ne(0U));
+
+    sut.portRouDi.releaseAllChunks();
+
+    // this is not part of the server port but holds the chunk from `sendResponse`
+    clientResponseQueue.clear();
+
+    EXPECT_THAT(getNumberOfUsedChunks(), Eq(0U));
+}
+
+// BEGIN tryGetCaProMessage tests
+
+TEST_F(ServerPort_test, TryGetCaProMessageOnOfferWhenNotOfferingHasCaProMessageTypeOffer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8944621b-4753-413b-bee0-a714fa4324c8");
+    auto& sut = serverPortWithoutOfferOnCreate;
+
+    sut.portUser.offer();
+
+    sut.portRouDi.tryGetCaProMessage()
+        .and_then([&](const auto& caproMessage) { EXPECT_THAT(caproMessage.m_type, Eq(CaproMessageType::OFFER)); })
+        .or_else([&]() { FAIL() << "Expected CaPro message but got none"; });
+}
+
+TEST_F(ServerPort_test, TryGetCaProMessageOnOfferWhenOfferingHasNullopt)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "15a399ee-b162-4b42-8aab-da13571fb478");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    sut.portUser.offer();
+
+    sut.portRouDi.tryGetCaProMessage()
+        .and_then(
+            [&](const auto& caproMessage) { FAIL() << "Expected no CaPro message but got: " << caproMessage.m_type; })
+        .or_else([&]() { SUCCEED(); });
+}
+
+TEST_F(ServerPort_test, TryGetCaProMessageOnStopOfferWhenOfferingHasCaProMessageTypeStopOffer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "83467e90-734b-4e51-836c-2dbeaf44ce95");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    sut.portUser.stopOffer();
+
+    sut.portRouDi.tryGetCaProMessage()
+        .and_then([&](const auto& caproMessage) { EXPECT_THAT(caproMessage.m_type, Eq(CaproMessageType::STOP_OFFER)); })
+        .or_else([&]() { FAIL() << "Expected CaPro message but got none"; });
+}
+
+TEST_F(ServerPort_test, TryGetCaProMessageOnStopOfferWhenNotOfferingHasNullopt)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a9a162d3-307a-4add-af23-63511da4b07e");
+    auto& sut = serverPortWithoutOfferOnCreate;
+
+    sut.portUser.stopOffer();
+
+    sut.portRouDi.tryGetCaProMessage()
+        .and_then(
+            [&](const auto& caproMessage) { FAIL() << "Expected no CaPro message but got: " << caproMessage.m_type; })
+        .or_else([&]() { SUCCEED(); });
+}
+
+// END tryGetCaProMessage tests
+
+// BEGIN test CaPro transitions
+
+TEST_F(ServerPort_test, StateNotOfferedWithCaProMessageTypeConnectReactsWithNack)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "ceaef856-2a8d-46c0-9167-fe1ca6fad736");
```

Comment:
Could we pack this and the following test cases until `StateNotOfferedWithCaProMessageTypeStopOfferReactsWithNack` into a function with `CaproMessageType` as argument?

Reply:
What do you think about an alternative like
```cpp
for(const auto caproMessageType: { CaproMessageType::CONNECT,
                                   CaproMessageType::DISCONNECT,
                                   CaproMessageType::STOP_OFFER})
{
    SCOPED_TRACE(caproMessageType);
    // here comes the test
}
```